### PR TITLE
Layout: modernize the layouts list

### PR DESCRIPTION
**Pull request by lenemter**
image

Changes:
- Use `Gtk.ListBox` instead of `Gtk.TreeView`
- Add margins to the items in the box
- Move "Remove", "Move up/down" buttons to the rows

### CODE REVIEW

File: src/Widgets/Layout/Display.vala [hunk 1]
```diff
@@ -40,50 +30,34 @@ public class Keyboard.LayoutPage.Display : Gtk.Frame {
...
         var scroll = new Gtk.ScrolledWindow (null, null) {
             hscrollbar_policy = Gtk.PolicyType.NEVER,
             expand = true
         };
-        scroll.add (tree);
-
-        add_button = new Gtk.Button.from_icon_name ("list-add-symbolic", Gtk.IconSize.BUTTON) {
-            tooltip_text = _("Add…")
-        };
+        scroll.add (list);
```
Comment (danirabbit) [inline src/Widgets/Layout/Display.vala:43]: while we're here can we use `child` property for Gtk4 future proofing?

File: src/Widgets/Layout/Display.vala [hunk 2]
```diff
@@ -96,110 +70,121 @@ public class Keyboard.LayoutPage.Display : Gtk.Frame {
             });
         });
 
-        remove_button.clicked.connect (() => {
-            settings.remove_active_layout ();
-            rebuild_list ();
-        });
+        list.row_activated.connect (() => {
+            warning ("Row activated");
```
Comment (danirabbit) [inline src/Widgets/Layout/Display.vala:74]: leftover from debugging?

File: src/Widgets/Layout/Display.vala [hunk 3]
```diff
@@ -96,110 +70,121 @@ public class Keyboard.LayoutPage.Display : Gtk.Frame {
...
+                };
+
+                var up_button = new Gtk.Button.from_icon_name ("go-up-symbolic") {
+                    tooltip_text = _("Move up"),
+                    halign = END
+                };
+
+                var down_button = new Gtk.Button.from_icon_name ("go-down-symbolic") {
+                    tooltip_text = _("Move down"),
+                    halign = END
```
Comment (danirabbit) [inline src/Widgets/Layout/Display.vala:148]: I don't think these `haligns` have any affect since you're already expanding the label

**Review by danirabbit — CHANGES_REQUESTED** [review]
Nice! Love actions on the rows themselves, much more clear which row will be affected.

It looks like the arrow buttons aren't being set as insensitive when an item is at the top or bottom of the list. For example:
Screenshot from 2023-08-17 10 04 03

Is there a reason you decided to ditch the list model? That can still be used with ListBox and would probably save some effort manually destroying and rebuilding the list when the model changes

**Comment by lenemter** [comment]
> Is there a reason you decided to ditch the list model?

I think the reason why it originally used a model is because of `Gtk.TreeView`, and it didn't save the code from manual rebuilding the tree :) I do agree that using the list model will save some effort but it can be added later if needed.

**Review by danirabbit — APPROVED** [review]
Nice work!

**Pull request merged**